server/kit/log.go

```diff
@@ -104,23 +105,58 @@ func LogMsg(ctx context.Context, message string) error {
 	return level.Info(Logger(ctx)).Log("message", message)
 }
 
+// Logf will format the given string with the arguments then log to the server logger
+// with the key "message" along with all the common request headers or gRPC metadata.
+// Arguments are handlered in the manner of fmt.Printf.
+// This message will have an "info" log level associated with it.
+func Logf(ctx context.Context, format string, v ...interface{}) error {
+	return level.Info(Logger(ctx)).Log("message", fmt.Sprintf(format, v...))
+}
+
 // LogDebug will log the given message to the server logger
 // with the key "message" along with all the common request headers or gRPC metadata.
 // This message will have a "debug" log level associated with it.
 func LogDebug(ctx context.Context, message string) error {
 	return level.Debug(Logger(ctx)).Log("message", message)
 }
 
+// LogDebugf will format the given string with the arguments then log to the server
+// logger with the key "message" along with all the common request headers or gRPC
+// metadata.
+// Arguments are handlered in the manner of fmt.Printf.
+// This message will have a "debug" log level associated with it.
+func LogDebugf(ctx context.Context, format string, v ...interface{}) error {
+	return level.Debug(Logger(ctx)).Log("message", fmt.Sprintf(format, v...))
+}
+
 // LogWarning will log the given message to the server logger
 // with the key "message" along with all the common request headers or gRPC metadata.
 // This message will have a "warn" log level associated with it.
 func LogWarning(ctx context.Context, message string) error {
 	return level.Warn(Logger(ctx)).Log("message", message)
 }
 
+// LogWarningf will the format given string with the arguments then log to the server
+// logger with the key "message" along with all the common request headers or gRPC
+// metadata.
+// Arguments are handlered in the manner of fmt.Printf.
+// This message will have a "warn" log level associated with it.
+func LogWarningf(ctx context.Context, format string, v ...interface{}) error {
+	return level.Warn(Logger(ctx)).Log("message", fmt.Sprintf(format, v...))
+}
+
 // LogErrorMsg will log the given error under the key "error", the given message under
 // the key "message" along with all the common request headers or gRPC metadata.
 // This message will have an "error" log level associated with it.
 func LogErrorMsg(ctx context.Context, err error, message string) error {
 	return level.Error(Logger(ctx)).Log("error", err, "message", message)
 }
+
+// LogErrorf will format the given string with the arguments then log to the server
+// logger with the key "message" along with all the common request headers or gRPC
+// metadata.
+// Arguments are handlered in the manner of fmt.Printf.
+// This message will have a "warn" log level associated with it.
+func LogErrorf(ctx context.Context, format string, v ...interface{}) error {
+	return level.Error(Logger(ctx)).Log("message", fmt.Sprintf(format, v...))
```

Comment:
Should this be `error` or `message`?  Not sure of difference between them.

Reply:
In general, I'd prefer folks to use `kit.LogErrorMsg` as it follows more of a "structured" logging approach.

This is meant to follow how the appengine/log's `log.Errorf`in that you only have a single string value to log. Under the hood, that logger puts the resulting string in the "message" JSON key for Stackdriver to consume. When viewing logs in the GCP console, the "message" value is actually whats displayed: https://cloud.google.com/logging/docs/view/overview#expanding

Reply:
TBH, with the current GCP logging UI I've been getting frustrated with the structured logging approach, though.

While viewing app logs within the combined request/app log UI, we're unable to _expand_ the app logs and actually view the error message. We have to go to app_logs only, re-find the log statement and then expand it to see the `error` field. To avoid that kind of pain this `LogErrorf` has everything in the main "message", which I'm tempted to start doing.

Reply:
I also find the UI frustrating.  I've been getting logs via command line, as I can get the actual relevant information.  I'm all for simplifying the logging!